### PR TITLE
samples/bluetooth/hci_uart: change bbc_microbit RAM demands to so test passes

### PR DESCRIPTION
**Pull request by pabigot**
The default test now requires in excess of 16 K RAM to store the
interrupt table.  Add the filter, and remove bbc_microbit (which
fails) from the whitelist.

Fixes build failure at https://app.shippable.com/github/zephyrproject-rtos/ci-test/runs/3154/3/tests

### CODE REVIEW

File: samples/bluetooth/hci_uart/sample.yaml [hunk 1]
```diff
@@ -3,13 +3,12 @@ sample:
   name: TBD
 tests:
   sample.bluetooth.hci_uart.arm:
+    min_ram: 17
```
Comment (carlescufi) [inline samples/bluetooth/hci_uart/sample.yaml]: I fixed this recently so it'd fit in 16KB of RAM. I'd really like to keep the bbc:microbit in the list of tested boards. Can you reduce the number of connections instead? See https://github.com/zephyrproject-rtos/zephyr/commit/9be6fb0ba275b62287632591374e35eb279b5a96

**Comment by aescolar** [comment]
@pabigot Do you know what caused the RAM use increase? (and why it was not detected in CI in that PR)

**Comment by pabigot** [comment]
> Do you know what caused the RAM use increase? (and why it was not detected in CI in that PR)

No, and no.  All I know is it failed in last nights `sanitycheck --all` run, and is the only test to have failed (though several matrix builds timed out, so more failures may be hidden).  The nightly test has not passed for about a week and I keep getting bit by unrelated failures like this, so I'm trying to fix it.

>  Can you reduce the number of connections instead?

Done.

**Comment by jhedberg** [comment]
It'd be good to understand what increased the footprint. This kind of gradual and uncontrolled upwards creep of it isn't particularly nice. Ideally we should have these memory-tight tests, like for the micro:bit, built on every PR so that footprint increases are caught immediately.

**Comment by pabigot** [comment]
It started failing at 6700f2f1940c645c5cd1eb23baad8bd0c8041b81 from #17293.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
> It'd be good to understand what increased the footprint. This kind of gradual and uncontrolled upwards creep of it isn't particularly nice. Ideally we should have these memory-tight tests, like for the micro:bit, built on every PR so that footprint increases are caught immediately.

The problem is that we keep adding features that are enabled by default, which increases the footprint of default builds.

**Comment by aescolar** [comment]
> The problem is that we keep adding features that are enabled by default, which increases the footprint of default builds.

But still, I really though this test in this board was built in each CI run by default.
@carlescufi Would you be so kind as to add it so we catch their like in the future?

**Comment by pabigot** [comment]
FWIW, these tests were run on `bbc_microbit` for that PR:
```
52523.1.log: 55/651 bbc_microbit              samples/boards/bbc_microbit/display/sample.board.bbc_microbit.display PASSED (build)
52523.1.log: 57/651 bbc_microbit              samples/boards/bbc_microbit/line_follower_robot/sample.board.bbc_microbit.line_follower_robot PASSED (build)
52523.1.log: 58/651 bbc_microbit              samples/bluetooth/ibeacon/sample.bluetooth.ibeacon PASSED (build)
52523.1.log: 59/651 bbc_microbit              samples/boards/bbc_microbit/sound/sample.board.bbc_microbit.sound PASSED (build)
52523.1.log: 60/651 bbc_microbit              samples/synchronization/sample.synchronization     PASSED (build)
52523.1.log: 61/651 bbc_microbit              samples/boards/bbc_microbit/pong/sample.board.bbc_microbit.pong PASSED (build)
52523.1.log: 64/651 bbc_microbit              tests/bluetooth/mesh/mesh.microbit                 PASSED (build)
```

so this test was not run in this case.

**Comment by carlescufi** [comment]
> > The problem is that we keep adding features that are enabled by default, which increases the footprint of default builds.
> 
> But still, I really though this test in this board was built in each CI run by default.
> @carlescufi Would you be so kind as to add it so we catch their like in the future?

I actually do not know how to do that. What defines the default set of tests that is run by CI by default?

**Review by jhedberg — APPROVED** [review]

**Comment by pabigot** [comment]
So this is amusing: This patch is not yet merged, but the nightly autotests are passing.

On master at 1ce95de6aee00f0433c45bb785807ce1dfe7ec4f we have:
```
tirzah[18]$ sanitycheck -T samples/bluetooth/hci_uart/
[...]
2 test configurations selected, 412 configurations discarded due to filters.
Adding tasks to the queue...
total complete:    2/   2  100%  skipped:    0, failed:    0
2 of 2 tests passed (100.00%), 0 failed, 0 skipped with 0 warnings in 8.06 seconds
In total 2 test cases were executed on 1 out of total 207 platforms (0.48%)
```
but
```
tirzah[16]$ sanitycheck -p bbc_microbit -T samples/bluetooth/hci_uart/
[...]
1 test configurations selected, 1 configurations discarded due to filters.
Adding tasks to the queue...
bbc_microbit              samples/bluetooth/hci_uart/sample.bluetooth.hci_uart.arm FAILED: Build failure
	see: sanity-out/bbc_microbit/samples/bluetooth/hci_uart/sample.bluetooth.hci_uart.arm/build.log
total complete:    1/   1  100%  skipped:    0, failed:    1
```

Looks like the new sanitycheck code ignores whitelist?  @carlescufi @nashif

**Comment by pabigot** [comment]
And shippable's [failing again](https://app.shippable.com/github/zephyrproject-rtos/ci-test/runs/3173/16/console) due to lack of this fix.

Can we merge this, or do we want to keep this failure present until somebody figures out how to make the test run in every sanitycheck?

**Pull request merged**